src/manifolds/ProbabilitySimplex.jl

```diff
@@ -0,0 +1,263 @@
+        return DomainError(
+            sum(p),
+            "The point $(p) does not lie on the $(M) since its sum is not 1.",
+        )
+    end
+    return nothing
+end
+
+"""
+    check_tangent_vector(M::ProbabilitySimplex, p, X; check_base_point = true, kwargs... )
+
+Check whether `X` is a tangent vector to `p` on the [`ProbabilitySimplex`](@ref) `M`, i.e.
+after [`check_manifold_point`](@ref)`(M,p)`, `X` has to be of same dimension as `p`
+and its elements have to sum to one.
+The optional parameter `check_base_point` indicates, whether to call
+[`check_manifold_point`](@ref)  for `p` or not.
+The tolerance for the last test can be set using the `kwargs...`.
+"""
+function check_tangent_vector(
+    M::ProbabilitySimplex,
+    p,
+    X;
+    check_base_point = true,
+    kwargs...,
+)
+    if check_base_point
+        mpe = check_manifold_point(M, p; kwargs...)
+        mpe === nothing || return mpe
+    end
+    mpv = invoke(
+        check_tangent_vector,
+        Tuple{typeof(get_embedding(M)), typeof(p), typeof(X)},
+        get_embedding(M),
+        p,
+        X;
+        check_base_point = false, # already checked above
+        kwargs...
+    )
+    mpv === nothing || return mpv
+    if !isapprox(sum(X), 0.0; kwargs...)
+        return DomainError(
+            sum(X),
+            "The vector $(X) is not a tangent vector to $(p) on $(M), since its elements to not sum up to 0.",
+        )
+    end
+    return nothing
+end
+
+decorated_manifold(M::ProbabilitySimplex) = Euclidean(representation_size(M)...; field = ℝ)
+
+@doc raw"""
+    distance(M,p,q)
+
+Compute the sistance of two points on the [`ProbabilitySimplex`](@ref) `M`.
+The formula reads
+````math
+d_{𝒮}(p,q) = 2\arccos \biggl( \sum_{i=1}^{n+1} \sqrt{p_i q_i} \biggr)
+````
+"""
+distance(::ProbabilitySimplex,p,q) = 2*acos(sum(sqrt.(p.*q)))
+
+embed!(M::ProbabilitySimplex, q, p) = (q .= p)
+embed!(M::ProbabilitySimplex, Y, p, X) = (Y .= X)
+
+@doc raw"""
+    exp(M::ProbabilitySimplex,p,X)
+
+Compute the exponential map on the probability simplex.
+
+````math
+\exp_pX = \frac{1}{2}\Bigl(p+\frac{1}{\lVert X_p \rVert^2}X_p^2\Bigr)
++ \frac{1}{2}\Bigl(p - \frac{1}{\lVert X_p \rVert^2}X_p^2\Bigr)\cos(\lVert X_p\rVert)
++ \frac{1}{\lVert Xp \rVert}\sqrt{p}\sin(\lVert X_p\rVert),
+````
+
+where $X_p = \frac{X}{\sqrt{p}}$, with its division meant elementwise, as well as for the
+operations $X_p^2$ and $\sqrt{p}$.
+"""
+exp(::ProbabilitySimplex, ::Any...)
+
+function exp!(::ProbabilitySimplex, q, p, X)
+    Xp = X./sqrt.(p)
+    Xp_n = Xp./norm(Xp)
+    q .= 0.5*(p+Xp_n.^2) + 0.5*(p-Xp_n.^2).*cos(norm(Xp)) + Xp_n.*sqrt.(p).*sin(norm(Xp))
+    return q
+end
+
+@doc raw"""
+    inner(M::ProbabilitySimplex,p,X,Y)
+
+Compute the inner product of two tangent vectors `X`, `Y` from the tangent space $T_p𝒮$ at
+`p`. The formula reads
+````math
+g_p(X,Y) = \sum_{i=1}^{n+1}\frac{X_iY_i}{p}
+````
+"""
+inner(::ProbabilitySimplex, p, X, Y) = sum( (X.*Y)./p )
```

Comment:
Like with `distance` above, this can be made more efficient by looping instead of broadcasting.